ui/window.go

```diff
@@ -885,6 +842,50 @@ func NewWindow(doRestart chan bool, app *tview.Application, session *discordgo.S
 	return window, nil
 }
 
+func (window *Window) TrySendMessage(message string) {
+	if window.selectedChannel == nil {
+		return
+	}
+
+	if len(message) == 0 {
+		if window.editingMessageID != nil {
+			msgIDCopy := *window.editingMessageID
+			window.askForMessageDeletion(msgIDCopy, true)
+		}
+		return
+	}
+
+	message = window.prepareMessage(message)
+	if len(message) > 2000 {
+		window.app.QueueUpdateDraw(func() {
+			window.ShowErrorDialog("Messages must be under 2000 characters to send")
+		})
+	} else if window.editingMessageID != nil {
+		go window.editMessage(window.selectedChannel.ID, *window.editingMessageID, message)
+		window.exitMessageEditMode()
+		window.OnMessageSent()
+	} else {
+		go func() {
+			messageText := window.jsEngine.OnMessageSend(message)
+			_, sendError := window.session.ChannelMessageSend(window.selectedChannel.ID, messageText)
+			window.chatView.internalTextView.ScrollToEnd()
```

Comment:
Move this into `sendError != nil`, as there shouldn't be a need to scroll if nothing has actually happened.